src/main/java/io/confluent/connect/jdbc/source/JdbcSourceTask.java

```diff
@@ -316,6 +328,8 @@ public void stop() throws ConnectException {
         return null;
       } catch (Throwable t) {
         resetAndRequeueHead(querier);
+        // This task has failed, so close any resources (may be reopened if needed) before throwing
+        closeResources();
```

Comment:
Maybe change this to `running.set(false)`, and then have a finally block around the contents of this method that calls `closeResources()`?  Then `closeResources()` would check that `running` is false before doing anything.

Reply:
Well `running` is more about the start-stopped state of the connector, not whether it is actually doing anything. The provider and dialect will both reopen if necessary, so that's not really an issue (tho they might not be closed again, which is kind of the state we might be in prior to this fix).